ipapython/ipautil.py

```diff
@@ -127,12 +127,34 @@ def __init__(self, addr):
         super(UnsafeIPAddress, self).__init__(addr,
                                               flags=self.netaddr_ip_flags)
 
+    def __getstate__(self):
+        state = {attr: getattr(self, attr) for attr in ['_net']}
+        try:
+            state['super_state'] = super(UnsafeIPAddress, self).__getstate__()
+        except AttributeError:
+            # none of base classes implements custom pickling
+            pass
+
+        return state
+
+    def __setstate__(self, state):
+        try:
+            super_state = state.pop('super_state')
+        except KeyError:
+            # no state saved for base classes
+            pass
+        else:
+            super(UnsafeIPAddress, self).__setstate__(super_state)
+
+        self.__dict__.update(state)
```

Comment:
You should be explicit about what you are unpickling, not only because explicit is better than implicit, but also for robustness - you can't guarantee the pickled data will contain only the attributes you want.